File: test_shopee_all/1.py
import pandas as pd

df = pd.read_csv('test_shopee_all/Customer.csv') #, names = ['customer_id', 'name', 'segment', 'city', 'state'], header = 1)
print(df.head(5))
# df.columns = ['customer_id', 'name', 'segment', 'city', 'state']
# print(df.head())

hasil = df.groupby('City')['Customer ID'].nunique().reset_index(name='jumlah').sort_values('jumlah', ascending=False).reset_index(drop=True)
print(hasil)

File: test_shopee_all/1.r
library(dplyr)

df <- read.csv('test_shopee_all/Customer.csv') #, col.names = c('customer_id', 'name', 'segment', 'city', 'state'))
print(head(df))
# colnames(df) <- c('customer_id', 'name', 'segment', 'city', 'state')
# print(head(df))

hasil <- df %>% group_by(City) %>% summarise(jumlah = n_distinct(Customer.ID)) %>% arrange(desc(jumlah))
print(head(hasil))
print(tail(hasil))